app/services/populate_email_addresses_table.rb

```diff
@@ -0,0 +1,39 @@
+class PopulateEmailAddressesTable
+  def initialize
+    @count = 0
+    @total = 0
+  end
+
+  # :reek:DuplicateMethodCall
+  def call
+    User.in_batches(of: 1000) do |relation|
+      sleep(1)
```

Comment:
1 second seems like a long time to sleep here. That adds a little of 2 hours of idle time to your job (assuming 8M users). With a batch size of 1000 you could probably do `0.25`.

Reply:
Wait, this looks like it isn't making any KMS calls. What is the reason for the limit here?

Reply:
The limit is to make sure we aren't driving the database too hard. This is the same process we went through when populating the phone configurations database. It does take about 6 hours to do everything, but I just let it run for the day.